### PR TITLE
docs: Add Lucide-Svelte icon import instructions and type definitions

### PR DESCRIPTION
**Pull request by shyakadavis**
## What is the purpose of this pull request?

- [ ] New Icon
- [ ] Bug fix
- [ ] New Feature
- [x] Documentation update
- [ ] Other:

### Description
Following the conversations in #1761 & #1585, this improves the docs around the areas of type definitions and import statements.

If there are any modifications you wish for me to carry out, please let me know. 🙂 

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by lolcabanon** [comment]
Great! Maybe a link to the Svelte doc to the ComponentType would be nice? Apart from that, it seems perfect to me.

Link : https://svelte.dev/docs/typescript#types-componenttype

File: docs/guide/packages/lucide-svelte.md [hunk 1]
```diff
@@ -71,6 +81,87 @@ To customize the appearance of an icon, you can pass custom properties as props
...
+  import type { ComponentType } from 'svelte';
+  import type { Icon } from 'lucide-svelte';
+
+  type Route = {
+    name: string;
+    href: string;
+    icon: ComponentType<Icon>;
+  }
+
+  const routes: Route[] = [
```
Comment (ericfennis) [inline docs/guide/packages/lucide-svelte.md:102]: For clearance nice to rename this to `menuItems`, I think that fits a bit better with what's happening in the component.

**Review by ericfennis — COMMENTED** [review]
Awesome thanks a lot, I've a small improvement for the examples

File: docs/guide/packages/lucide-svelte.md (hunk 1 above)
Comment (shyakadavis) [reply docs/guide/packages/lucide-svelte.md]: Done, @ericfennis 

I've also made a minor update about the manner of import to use the recommended/fast way.

File: docs/guide/packages/lucide-svelte.md [hunk 2]
```diff
@@ -71,6 +81,91 @@ To customize the appearance of an icon, you can pass custom properties as props
...
+  import type { ComponentType } from 'svelte';
+  import type { Icon } from 'lucide-svelte';
+
+  type Route = {
+    name: string;
+    href: string;
+    icon: ComponentType<Icon>;
+  }
+
+  const menuItems: Route[] = [
```
Comment (ericfennis) [inline docs/guide/packages/lucide-svelte.md:104]:
```suggestion
  type MenuItem = {
    name: string;
    href: string;
    icon: ComponentType<Icon>;
  }

  const menuItems: MenuItem[] = [
```

File: docs/guide/packages/lucide-svelte.md [hunk 3]
```diff
@@ -71,6 +81,91 @@ To customize the appearance of an icon, you can pass custom properties as props
...
+
+  /**
+   * @typedef {Object} Route
+   * @property {string} name
+   * @property {string} href
+   * @property {import('svelte').ComponentType<import('lucide-svelte').Icon>} icon
+   */
+
+  /** @type {Route[]} */
+  const menuItems = [
```
Comment (ericfennis) [inline docs/guide/packages/lucide-svelte.md:147]:
```suggestion
  /**
   * @typedef {Object} MenuItem
   * @property {string} name
   * @property {string} href
   * @property {import('svelte').ComponentType<import('lucide-svelte').Icon>} icon
   */

  /** @type {MenuItem[]} */
  const menuItems = [
```

**Review by ericfennis — COMMENTED** [review]
Also rename the types?

**Review by ericfennis — APPROVED** [review]
Awesome work!

**Pull request merged**